contrib/internal/httputil/trace.go

```diff
@@ -54,6 +58,34 @@ func TraceAndServe(h http.Handler, cfg *TraceConfig) {
 
 	cfg.ResponseWriter = wrapResponseWriter(cfg.ResponseWriter, span)
 
+	op := dyngo.StartOperation(
+		httpinstr.HandlerOperationArgs{
+			IsTLS:       cfg.Request.TLS != nil,
+			Method:      httpinstr.Method(cfg.Request.Method),
+			Host:        httpinstr.Host(cfg.Request.Host),
+			RequestURI:  httpinstr.RequestURI(cfg.Request.RequestURI),
+			RemoteAddr:  httpinstr.RemoteAddr(cfg.Request.RemoteAddr),
+			Headers:     httpinstr.Header(cfg.Request.Header),
+			UserAgent:   httpinstr.UserAgent(cfg.Request.UserAgent()),
+			QueryValues: httpinstr.QueryValues(cfg.Request.URL.Query()),
+		},
+	)
+	op.OnData(func(_ *dyngo.Operation, e *appsectypes.SecurityEvent) {
```

Comment:
1. Not in the RFC because I wanted to see if I could avoid it at the time I wrote it. Now I am 100% convinced we need them as they allow to emit "dynamic events" in an operation lifetime. I should add it to the RFC now.
2. It works exactly like the start and finish events: according to the emitted/listened types.

So indeed what you see in `appsec/internal/protection/http/http.go` is there to listen for security events possibly happening in the http operation lifetime (for now when the WAF triggers).

I didn't want to introduce buisness-specific APIs to dyngo, so I avoided `OnSecurityEvent` as it could be implemented with the data event necessary for the WAF to subscribe to the addresses its rule uses.

Does it make sense?